File: SuperNewRoles/Map/ImageManager.cs
using System;
using System.Collections.Generic;
using System.Text;
using UnityEngine;

namespace SuperNewRoles.Map
{
    public static class ImageManager
    {
        private static Sprite DoorSabotageButton;
        public static Sprite GetDoorSabotageButton()
        {
            if (DoorSabotageButton != null) return DoorSabotageButton;
            DoorSabotageButton = ModHelpers.loadSpriteFromResources("SuperNewRoles.Resources.DoorSabotageButton.png", 115f);
            return DoorSabotageButton;
        }
    }
}

File: cpp/Map/ImageManager.cpp
#include "ImageManager.h"
#include "../ModHelpers.h"

using namespace UnityEngine;

namespace SuperNewRoles::Map
{

Sprite *ImageManager::AgarthaAdminButton;

	Sprite *ImageManager::GetAgarthaAdminButton()
	{
		if (AgarthaAdminButton != nullptr)
		{
			return AgarthaAdminButton;
		}
		AgarthaAdminButton = ModHelpers::loadSpriteFromResources(L"SuperNewRoles.Resources.Agartha.AdminButton.png", 115.0f);
		return AgarthaAdminButton;
	}
}
